### DISCUSSION TITLE
question about process for ordering PCBs

### DISCUSSION THREAD

**Discussion by maks** [xiphonics/picoTracker-pcb#3] [General]
Hi, thanks for making available all your hard work on this brillant project!

I'm a [fan of lgpt myself](https://www.youtube.com/shorts/87MJ3PwhKKY) and would love to have a go at building the Picotracker. 

When I first came across your project yesterday I was going to have a go at building the original version as I realised I had everything in my spareparts box bar the lcd (which I ordered) but then I saw your new portable version of Picotracker and I would definitely love to build that instead. I read through your [description of the build](https://democloid.com/2023/06/22/picoTrackerPortable.html) and it seems I would need just the LCD part vs the "module" version which comes on its own pcb board, is that right? 

Also could I ask for some advice on whats required to order a assembled version of the PCB? I've ordered just bare PCBs before from the likes of PCBway by just uploading the Gerber which was simple enough, but I'm thinking ordering preassembled boards is a bit more involved? And I'm guessing they only do the SMD parts and that I still need to solder in the 3.5mm, usb & sdcard sockets myself?

**Comment by democloid** [comment]
Hey, glad you like it and are willing to build it! I'd recommend doing the RP pico version first to check if this works for you (specially if you have all the parts), there are some things that are a little janky right now, I've been procrastinating releasing a new version to fix those.

Regarding the screen module, yes, this is the exact module I bought: https://www.aliexpress.us/item/2255799833458861.html

For JLCPCB (which is what I used) you have all the necessary files in the release section. For assembly service you just need the BOM and the pos files added to the gerber files. After you add your gerber files, you need to go to the lower section where is says "PCB Assembly", enable it and it will ask what side to assemble, top side is the default and what we want. You can choose how many to assemble, 2 is the minimum. After you click confirm you're presented with a screen where you can upload BOM and POS (or CPL) files. After this you'd have to check "component placements", this part is a bit trickier because you may have some parts that are not correctly positioned, It's usually obvious and there are some "autocorrect" tools, but let me know if something doesn't look right. I should do a guide to explain all this process. 

Regarding your question about parts, this will give you a completely assembled PCB, connectors, SD card slot, USB, etc. (This is exactly what I received from JLCPCB https://democloid.com/assets/img/2023-6-22-picoTrackerPortable/pcb_real.jpg). The only things that need to be fitted are the screen and (optional) battery.

Let me know if there is anything else I can help with.

**Comment by maks** [comment]
Brilliant, thanks for such a quick and detailed reply!

Yes thats a good point, I'll have at just making the initial model first to better learn the process of putting it all together before I tackle the portable model. And thanks also for running me through the process of ordering the PCB, that all makes sense and very helpful! I think I may be in similar boat to you when you started, I did elec eng so I should know all this, but over 25 years ago which makes memory quite hazy.

I'll get the order in though as shipping times to here in Aus are often quite lengthy, or did you mean that you have an updated version of the PCB that I should wait for before ordering?

Thanks again!

**Comment by maks** [comment]
Hi I did the easy thing first and ran the case print on my printer (the case design is *very* nice!) and I wanted to get a sense for if the quiet linear keys I have already from an [old crowdfunder](https://www.crowdsupply.com/open-steno-project/quiet-linear-mechanical-keyboard-switch) along with some left over blank keycaps I have would work and it seems that they will fit. Apologies its not in the official purple.

I think I will need to try to make a modified version of the keys PCB as these switches have [a different footprint](https://github.com/KiCad/Buttons_Switches_THT.pretty/blob/master/SW_Matias_Quiet.kicad_mod) to the MX ones or if are have the chance, would you have time to make a second PCB layout with them as I have to admit I've never used KiCAD before.

Super excited to try to build the "desktop" version once the LCD I ordered locally gets here next week and then the portable once the various parts and PCBs once I order them arrive.

![PXL_20231002_032712778](https://github.com/democloid/picoTrackerPortable-pcb/assets/71999/c5f7152d-7bda-478f-8770-d8d691e1c3dc)

**Comment by democloid** [comment]
PCB should be final, no concerns there. Only problem with small runs is that it may be significantly more expensive per part.

Love the color combination, I'd even encourage uniqueness through different colors :)

Reason I chose the cherry low profile switches is that otherwise the whole case would be way too thick and not look good (to me). 10 pcs of these will be less than 10 bucks in aliexpress, so it's not too bad, keycaps that can be used with them are limited tho, DSA profile ones work well. Can also 3D print the keycaps but I found that not to work too well.

**Comment by maks** [comment]
Ah great thanks! I've placed the PCB order now, thanks again for running through it, especially the step to fix up the placements (seemed to mostly just be rotations which were easy given the pin-1 dots on the pcb).

And yes I see now what you mean about low profile switches. I guess I can just use up my existing ones on the "desktop build". With low profiles switches, is [something like these](https://www.aliexpress.com/item/1005005818694764.html?spm=a2g0o.productlist.main.1.383b7d83r34ijd&algo_pvid=16fc19fa-0021-4952-a1fb-edcdc1971b21&algo_exp_id=16fc19fa-0021-4952-a1fb-edcdc1971b21-0&pdp_npi=4%40dis%21AUD%2139.93%2119.57%21%21%21182.84%21%21%402101f4a016963053203737092e35b3%2112000034477790068%21sea%21AU%21178034516%21&curPageLogUid=cyvpcKQJXDT1) that you used?

Also would you be open to opening the Github discussions feature on this or one of your other repos as that might be better than my continuing to comment on this issue, since you've more than answered it so it could be closed now 🙂

**Comment by maks** [comment]
I surprised myself by being able to rustle up everything bar the screen from spare parts box, though the sd board is not actually the adafruit one, so I just need to remove the little studs on the case to make it fit perfectly with a bit of hot glue on it and the switches.
![build-prep](https://github.com/democloid/picoTrackerPortable-pcb/assets/71999/952dd2c9-b3a9-456b-825e-62d877d16b56)

**Reply by maks** [reply]
@democloid thanks so much for setting up the discussions, I felt a bit bad as a software dev just posting all my build progress comments and photos to the issue. 
Unfortuately I think the seller on ebay I bought my LCD from is drop shipping so I likely wont get it til next week, but I'm keen to start soldering up all the other parts this weekend and very excited to get the picotracker built and use it.

**Comment by democloid** [comment]
Nice!! Eager to know how it goes. I'll look into the discussions feature, thanks.

Switches are these: https://www.aliexpress.us/item/3256804327585812.html

**Reply by maks** [reply]
Perfect thanks for that! That really helps to know what to look for🙏🏻 
Doing a search, I found similar ones for a fair bit cheaper [at a local shop](https://www.scorptec.com.au/product/keyboards/keyboard-switches/104352-) (per 100 anyways) so I can get them without having to wait a couple of weeks shipping from china 🙂 but it will likely be a couple of weeks til I get the PCBs and lcds, batteries from ali.

**Reply by maks** [reply]
@democloid btw just as an fyi, I think prices may have gone up a bit since your order as this is the parts price list I got on my order. This prices are in AUD not USD, but it still after coupons came out at around $aud30 per board, which honestly still is very reasonable to me given I dont need to do any SMT soldering! 😅  


[JLCPCB-picotracker-BOM-costs.pdf](https://github.com/democloid/picoTrackerPortable-pcb/files/12797762/JLCPCB-picotracker-BOM-costs.pdf)

**Reply by democloid** [reply]
I see no total there so I'll believe you. I think my price was 110 USD for the whole order of 5 bords. I also calculated that doing 50 boards would be around 11 USD per board, which is pretty good.

**Reply by maks** [reply]
Oh sorry about that, the total for me was about $150AUD with shipping for 5 boards after coupon I got as a new customer, so actually that does work about same as yours, given the AUD exchange rate is a bit down lately. 
So I think thats my mistake! I saw in your blog post that it was around $22 but I misread that as for the 5 boards instead of per board, all makes sense now.
Hopefully I'll get mine in a week or 2 and can then move onto building them 🙂

**Reply by maks** [reply]
Yikes, I just discovered (luckily before ordering any) that all the brands low-profile switches seem to have incompatible footprints, unlike the fullsize ones so to be safe I just ordered a pack of the ones you linked to @democloid . On the topic of keys, can I check are the keycaps the same as the ones you linked to for the original "desktop" picoTracker?

**Reply by democloid** [reply]
Not all keycaps work on those low profile switches, almost sure the ones I used for the desktop version wouldn't (and it wouldn't look nice either due to height). What I used and I know to work are DSA profile switches: https://www.aliexpress.us/item/3256803662982415.html

There are some 3D printing options you can try too:
https://www.thingiverse.com/thing:5180641

https://www.thingiverse.com/thing:3311616

While I liked the look of the 1st option, I felt it was too wobbly/imprecise and they would touch each other. Maybe your 3D printer produces better results tho, and easy to try.

**Reply by maks** [reply]
Awesome thanks again! 🙏🏻 I'm lucky to have recently got a new Bambu P1P which is not only much faster but much more accurate than my ancient old little printer so I'll have a go at both of those as the designs look quite nice! But I'll also order some of those DSA caps as a comparison point and fallback if the 3d prints dont work out.

**Comment by maks** [comment]
Sorry to bug you again @democloid but I got an email from JLCPCB to ask if the "the polarities of the components SW1 are correct or not."
I guess I could have got the switch orientation wrong as I left it as autoplaced since I wasn't sure which way it went based on the pcb markings. Not a big deal as I think worse case it will just reverse the on/off direction but if you have a chance to quickly cast your eye over the layout below they sent me that would be great.

![Produce_DanZhi SMT_Snapshot Top 6650429A_Y3 SMT02310031417356](https://github.com/democloid/picoTrackerPortable-pcb/assets/71999/25daa7a4-cc1b-47b2-9acd-38c4454a1853)

**Reply by democloid** [reply]
hmmm, it's a simple toggle switch, there is no polarity I think? The on/off direction is really determined by the connections on the board, lower pins are floating so down is off.

I gave a quick glance at the rest of the components and all looks good to me, the main thing to look for is pin zero of components to the marking on the PCB, and polarity of polarized capacitors to their respective markings on the PCB too.

**Reply by maks** [reply]
Thanks so much for giving it a look over, much appreciated! Ah right yes that makes sense with the switch and good point about the capacitors, I didnt really check them, I only checked all the pin dots matched the printed ones on the pcb. Unfortunately my elec eng uni days are also a good 20+ years behind me so I'm *very* rusty on my electronics having only done software since then.
